// @valiron/sdk/dist/clients.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValironClient = void 0;
const types_1 = require("./types");
/**
 * HTTP client for the Valiron Operator API.
 *
 * All methods map 1-to-1 to the /operator/* endpoints exposed by the
 * Valiron edge-proxy service.
 */
class ValironClient {
    constructor(config = {}) {
        this.endpoint =
            config.endpoint || "https://valiron-edge-proxy.onrender.com";
        this.apiKey = config.apiKey;
        this.defaultChain = config.chain;
        this.timeout = config.timeout || 5000;
        this.debug = config.debug || false;
    }
    // ─── Public API ─────────────────────────────────────────────────────────
    /** Resolve the effective chain: per-call override > constructor default > undefined (server default). */
    resolveChain(options) {
        return options?.chain ?? this.defaultChain;
    }
    /**
     * Get the complete trust profile for an agent.
     *
     * Combines ERC-8004 on-chain identity, on-chain reputation, local
     * behavioral reputation, and a hybrid routing decision.
     *
     * `GET /operator/agent/:agentId`
     */
    async getAgentProfile(agentId, options) {
        return this.get(`/operator/agent/${agentId}`, this.resolveChain(options));
    }
    /**
     * Get a trust profile by wallet address (reverse lookup).
     *
     * `GET /operator/wallet/:wallet`
     */
    async getWalletProfile(wallet, options) {
        return this.get(`/operator/wallet/${wallet}`, this.resolveChain(options));
    }
    /**
     * Trigger real sandbox testing for an agent.
     *
     * Sends test requests against the agent's HTTP endpoint (if one is
     * registered on-chain) or through the sandbox-api relay, then computes
     * a Valiron score and stores the result.
     *
     * `POST /operator/trigger-sandbox/:agentId`
     */
    async triggerSandboxTest(agentId, options) {
        return this.post(`/operator/trigger-sandbox/${agentId}`, undefined, this.resolveChain(options));
    }
    // ─── HTTP helpers ───────────────────────────────────────────────────────
    buildHeaders() {
        const headers = {
            "Content-Type": "application/json",
        };
        if (this.apiKey) {
            headers["Authorization"] = `Bearer ${this.apiKey}`;
        }
        return headers;
    }
    async get(path, chain) {
        const url = new URL(path, this.endpoint);
        if (chain)
            url.searchParams.set("chain", chain);
        if (this.debug) {
            console.log(`[Valiron SDK] GET ${url.toString()}`);
        }
        const controller = new AbortController();
        const timeoutId = setTimeout(() => controller.abort(), this.timeout);
        try {
            const response = await fetch(url.toString(), {
                method: "GET",
                headers: this.buildHeaders(),
                signal: controller.signal,
            });
            clearTimeout(timeoutId);
            if (!response.ok) {
                const body = await response
                    .json()
                    .catch(() => ({ message: response.statusText }));
                throw new types_1.ValironError(body.error || body.message || `Request failed: ${response.statusText}`, "API_ERROR", response.status);
            }
            const data = await response.json();
            if (this.debug)
                console.log(`[Valiron SDK] Response:`, data);
            return data;
        }
        catch (error) {
            clearTimeout(timeoutId);
            if (error.name === "AbortError") {
                throw new types_1.ValironError(`Request timeout after ${this.timeout}ms`, "TIMEOUT_ERROR");
            }
            if (error instanceof types_1.ValironError)
                throw error;
            throw new types_1.ValironError(error.message || "Network error", "NETWORK_ERROR");
        }
    }
    async post(path, body, chain) {
        const url = new URL(path, this.endpoint);
        if (chain)
            url.searchParams.set("chain", chain);
        if (this.debug) {
            console.log(`[Valiron SDK] POST ${url.toString()}`, body);
        }
        const controller = new AbortController();
        const timeoutId = setTimeout(() => controller.abort(), this.timeout);
        try {
            const response = await fetch(url.toString(), {
                method: "POST",
                headers: this.buildHeaders(),
                body: body ? JSON.stringify(body) : undefined,
                signal: controller.signal,
            });
            clearTimeout(timeoutId);
            if (!response.ok) {
                const respBody = await response
                    .json()
                    .catch(() => ({ message: response.statusText }));
                throw new types_1.ValironError(respBody.error || respBody.message || `Request failed: ${response.statusText}`, "API_ERROR", response.status);
            }
            const data = await response.json();
            if (this.debug)
                console.log(`[Valiron SDK] Response:`, data);
            return data;
        }
        catch (error) {
            clearTimeout(timeoutId);
            if (error.name === "AbortError") {
                throw new types_1.ValironError(`Request timeout after ${this.timeout}ms`, "TIMEOUT_ERROR");
            }
            if (error instanceof types_1.ValironError)
                throw error;
            throw new types_1.ValironError(error.message || "Network error", "NETWORK_ERROR");
        }
    }
}
exports.ValironClient = ValironClient;
