// @valiron/sdk/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValironClient = exports.ValironSDK = void 0;
const clients_1 = require("./clients");
/**
 * Valiron SDK – Official TypeScript SDK for the Trust Layer for Agentic Economy
 *
 * @example
 * ```typescript
 * import { ValironSDK } from '@valiron/sdk';
 *
 * // Default chain (ethereum)
 * const valiron = new ValironSDK();
 *
 * // Or set a default chain at construction
 * const monadValiron = new ValironSDK({ chain: 'monad' });
 *
 * // Get full trust profile (identity + reputation + routing)
 * const profile = await valiron.getAgentProfile('25459');
 *
 * // Override chain per-call
 * const monadProfile = await valiron.getAgentProfile('8348', { chain: 'monad' });
 *
 * if (profile.routing.finalRoute === 'prod') {
 *   // Allow full production access
 * }
 * ```
 */
class ValironSDK {
    constructor(config = {}) {
        this.client = new clients_1.ValironClient(config);
    }
    /**
     * Get the complete trust profile for an ERC-8004 agent.
     *
     * Returns on-chain identity, on-chain reputation (feedback count,
     * average score, individual entries), local behavioral reputation
     * (Valiron sandbox metrics), and a hybrid routing decision.
     *
     * @param agentId - The ERC-8004 agent ID
     * @param options - Optional overrides (e.g. `{ chain: 'monad' }`)
     * @returns Full agent trust profile
     *
     * @example
     * ```typescript
     * // Uses default chain
     * const profile = await valiron.getAgentProfile('25459');
     *
     * // Explicit Monad chain
     * const monadProfile = await valiron.getAgentProfile('8348', { chain: 'monad' });
     *
     * console.log(profile.routing.finalRoute);   // 'prod'
     * console.log(profile.routing.decision);      // human-readable explanation
     * console.log(profile.localReputation?.tier);  // 'AAA'
     * console.log(profile.onchainReputation.averageScore); // 92.5
     * ```
     */
    async getAgentProfile(agentId, options) {
        return this.client.getAgentProfile(agentId, options);
    }
    /**
     * Convenience helper – get just the routing decision for an agent.
     *
     * This calls `getAgentProfile` under the hood and extracts the
     * `routing.finalRoute` field.
     *
     * @param agentId - The ERC-8004 agent ID
     * @param options - Optional overrides (e.g. `{ chain: 'monad' }`)
     * @returns The route decision string
     *
     * @example
     * ```typescript
     * const route = await valiron.checkAgent('25459');
     * const monadRoute = await valiron.checkAgent('8348', { chain: 'monad' });
     * if (route === 'prod' || route === 'prod_throttled') {
     *   return handleRequest();
     * }
     * ```
     */
    async checkAgent(agentId, options) {
        const profile = await this.client.getAgentProfile(agentId, options);
        return profile.routing.finalRoute;
    }
    /**
     * Get a trust profile by wallet address (reverse lookup).
     *
     * If the wallet has an associated agent ID in local state, the full
     * profile (identity + on-chain reputation) is included. Otherwise
     * only local behavioral data is returned.
     *
     * @param wallet - Ethereum wallet address
     * @param options - Optional overrides (e.g. `{ chain: 'monad' }`)
     * @returns Wallet trust profile
     *
     * @example
     * ```typescript
     * const profile = await valiron.getWalletProfile('0x52ce…');
     * console.log(profile.routing.finalRoute);
     * ```
     */
    async getWalletProfile(wallet, options) {
        return this.client.getWalletProfile(wallet, options);
    }
    /**
     * Trigger real sandbox testing for an agent.
     *
     * Sends a series of test requests against the agent's HTTP endpoint
     * (if registered on-chain) or through the sandbox relay, then computes
     * a Valiron behavioral score and stores the result.
     *
     * @param agentId - The ERC-8004 agent ID
     * @param options - Optional overrides (e.g. `{ chain: 'monad' }`)
     * @returns Sandbox test results including score, tier, and metrics
     *
     * @example
     * ```typescript
     * const result = await valiron.triggerSandboxTest('25459');
     * console.log(result.valironScore); // 95
     * console.log(result.tier);         // 'AAA'
     * console.log(result.riskLevel);    // 'GREEN'
     * console.log(result.testSummary);  // { totalRequests, successCount, … }
     * ```
     */
    async triggerSandboxTest(agentId, options) {
        return this.client.triggerSandboxTest(agentId, options);
    }
}
exports.ValironSDK = ValironSDK;
// Re-export everything
__exportStar(require("./types"), exports);
var clients_2 = require("./clients");
Object.defineProperty(exports, "ValironClient", { enumerable: true, get: function () { return clients_2.ValironClient; } });
